### DISCUSSION TITLE
Full met.hu weather integration

### DISCUSSION THREAD

**Discussion by nagyrobi** [amaximus/met_alerts_hu#4] [Ideas]
Would be great to have a full met.hu weather integration, combined with these alerts too!

Currently I combine openwheathermap and met.no with met_alerts_hu in order to get in one place:
- alerts from here
- hourly forecast from openwheathermap  (`onecall_daily`) for hourly forecasts on the same day
- daily forecast from met.no (HA's default) for min/max temperatures and conditions for the next few days

Woulbe better to have all these from met.hu...

![kép](https://user-images.githubusercontent.com/1550668/122775750-bfdae880-d2aa-11eb-9c65-b52233c4f9bd.png) ![kép](https://user-images.githubusercontent.com/1550668/122775912-e862e280-d2aa-11eb-989e-b8ff080b0d18.png)

(displayed with [openHASP](https://haswitchplate.github.io/openHASP-docs/0.6/) on Lanbon L8)

**Comment by amaximus** [comment]
Well, this integration was specifically created for meteo alerts as reported by met.hu. This is not a full meteorological integration as met.hu does not provide an API, so html pages have to be manipulated all the time.

There are so many weather integrations out there providing weather forecast and all sort of meteorological data (wind intensity and direction, UV-index, etc.).  I'm using darksky platform with custom weather-card-chart card [1].
[1] https://github.com/Yevgenium/lovelace-weather-card-chart

**Reply by nagyrobi** [reply]
Yes I know. I just thought that having a nice detailed met.hu local source.

**Reply by Anroxo** [reply]
This might be helpful. https://odp.met.hu/

**Reply by amaximus** [reply]
Thanks, I didn't know about this. I'm pretty sure these data are processed and presented on the met.hu webpage.
It would be good if met.hu had some sort of API.

**Comment by sinexton** [comment]
Hi Robi, if you allow me to suggest an idea, I think it might be also interesting to see a summary by a single information for the whole week that answers the question: What temperature range can we expect for the whole week as we would feel?

Usually as humans we need to interpret at least 14 numbers to answer this (min and max temp for 7 days). Math.hu has 28 numbers for this because they show a low and high limit for both the daily min and max values as well. That's a sea of numbers for the human brain. I'm using a math solution of mine for this and I thought I'd share with you that. It has statistical backup.

For myself I generate 2 numbers for the whole week representing the min and max temp that will be general for the week. For the min value I subtract the standard error from the mean of all min and max values of the week. And for the max value I add the double of se to the mean. I use single se for the min point because it is early at dawn that we meet that less than the high point of the day which may be around 12 or 14 o'clock.

An example: Let's say that we have the following max and min values for the next 7 days as a forecast:

26 26 25 16 13 12 13
11 14 15 12 8 7 7

m = the mean (average) of all the numbers above = 14.6
se = standard error of all the numbers above = 1.8

I round the result to integer values to help human interpretation.
minimum = m - se = 13
maximum = m + se * 2 = 18

So the answer is that we can expect the temperature between 13 and 18 degrees by feeling for the whole week.

If you like it and have any questions, let me know.

Andras

**Reply by nagyrobi** [reply]
Interesting.

Did you implement this in HA?

**Reply by sinexton** [reply]
I'm not familiar with HA. Just wanted to suggest you this calculation because it is easy to implement (simple statistics) and it has high power because it considers the fluctuation and so the expected long term lower and higher limit.
